test/utils/net.go

```diff
@@ -5,97 +5,98 @@ import (
 	"net"
 	"testing"
 	"time"
-
-	"github.com/stretchr/testify/require"
 )
 
-// WaitServiceOnline dials port(s)to check if the service comes online until it reaches the maximum retry
+const dialTimeout = 2 * time.Second
+
+// WaitServiceOnline waits for a service to come online by dialing its port(s)
+// up to a maximum number
 func WaitServiceOnline(t *testing.T, ports ...string) {
-	const dialTimeout = 2 * time.Second
 	const maxRetry = 60
 
+PORTS:
 	for _, port := range ports {
-		serviceIsOnline := false
 		var returnErr error
 
-		for i := 0; !serviceIsOnline && i < maxRetry; i++ {
-			t.Logf("Waiting for service. Dialing port %s. Retry %d/%d", port, i+1, maxRetry)
-			conn, err := net.DialTimeout("tcp", net.JoinHostPort("127.0.0.1", port), dialTimeout)
+		for i := 1; i <= maxRetry; i++ {
+			t.Logf("Waiting for service port %s. Retry %d/%d", port, i, maxRetry)
+
+			conn, err := net.DialTimeout("tcp", ":"+port, dialTimeout)
 			if conn != nil {
-				serviceIsOnline = true
-				t.Logf("Service online now. Port %s is listening", port)
+				t.Logf("Service port %s is open.", port)
+				continue PORTS
 			}
 			returnErr = err
+
 			time.Sleep(1 * time.Second)
 		}
 
-		require.Equal(t, true, serviceIsOnline,
-			"Service timed out, reached max %d retries. Error:\n%v", maxRetry, returnErr)
+		if returnErr != nil {
+			t.Fatalf("Time out: reached max %d retries. Error: %v", maxRetry, returnErr)
+		} else {
+			t.Fatalf("Time out: reached max %d retries.", maxRetry)
+		}
 	}
 }
 
-// PortConnection checks if the port(s) are in use
-func PortConnection(host string, ports ...string) (bool, error) {
-	var isListening bool
-	var err error
+// RequirePortOpen checks if the local port(s) accepts connections
+func RequirePortOpen(t *testing.T, host string, ports ...string) {
```

Comment:
Has `host` been used in this `RequirePortOpen()` function?

Reply:
Removed. Strangely, the callers were not passing it and tests were not failing. First port was being passed as host. I added checks so that this won't get passed the tests anymore.